blog/_posts/2017-09-12-discovering-discourse.md

```diff
@@ -0,0 +1,50 @@
+---
+title: 💎 D for diamonds are forever
+layout: post
+created_at: Tue September 12 2017
+permalink: blog/2017-09-12-d-for-diamonds
+current: blog
+author: Jen
+twitter: jenlijo
+categories: student-posts
+---
+## D for Discourse
+
+It's incredible how fast the summer has gone! (literally as well in Berlin). Choosing a project wasn't easy at first. After researching and consulting our coaches, Discourse seemed the right one to go. In the beginning it was an unknown for us and soon we would discover the immensity of it! A big and beautiful deep ocean of Rails and Ember code hanging around the internet in every discuss corner.
+
+Discourse is great to start to contribute to open source. The community in http://meta.discourse.org is willing to help and answer any question you might have. It comes full of information to get to know the project and a [Beginner's tutorial](https://meta.discourse.org/t/beginners-guide-to-creating-discourse-plugins-part-1/30515) to build something locally and get familiar with their particular way to add plugins to the main app.
+
+Kaja has also written [some posts in our blog](https://berlindiamonds.blogspot.de/) about what it is to contribute here, and we will try to add up more info to make it a complete guide.
+
+## What we built
+
+Our project was based on creating plugins to let the admins back up the information of the forum in some cloud provider.
+
+We didn't know how long it would take us to build one plugin. I thought we would have one or two at the end of the summer, but we ended up [enabling 4 possibilities](https://meta.discourse.org/t/new-plugins-for-backups/68018) as well as a base class plugin from which all of them inherit (Google Drive, Nextcloud - Jan from Nextcloud left a message in our repo suggesting it, we were super happy to get it and built it right away! - Box, and a new version of the existent backups to Dropbox by @falco). After this, our mentor @eviltrout found another cool task to give us: downloading these files directly from Discourse and send an email to the user when it's done, which we are on the way of building.
+
+![Team berlindiamonds](/img/blog/2017/2017-09-12-berlindiamonds.png)
+<div class="image-credits">Github screenshot</div>
+
+We've learnt so many things this summer!: inheritance, injecting code through class_eval and then through event listeners, git git git, a lot of interacting with APIs, the power of gems, how they're built, code structure, the "super" keyword, memoization, triggering Jobs, executing stuff in the console, debugging with puts, byebug, Rails.logger.debug, some testing, drawing routes from the outside, Ember basics... the importance of proper documentation!
+
+We've also participated in a RG beginners workshop in August, Kaja is part of the organization and proposed me to coach. It was a great feeling to see so many women (around 40) curious to code and many coaches happy to share their skills.
```

Comment:
Change 'we've also participated' -> '**we** also participated' 
Add a name here in brackets: Kaja is part of the organization and proposed me ['add your name'] to coach